runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -5950,6 +5951,263 @@ static TR::Register *inlineCRC32CUpdateBytes(TR::Node *node, TR::CodeGenerator *
+    *    addv vtmp1Reg, vtmp1Reg.4s
+    *    mov resultReg, vtmp1Reg.s[0]
+    *    addws lengthReg, lengthReg, #(8 or 16)
+    *    b.eq done
+    * residual:
+    *    subws lengthReg, lengthReg, #1
+    * if decompressed
+    *    ldrh dataReg, [dataAddrReg], #2
+    * else ; compressed
+    *    ldrb dataReg, [dataAddrReg], #1
+    * fi
+    *    addw dataReg, dataReg, resultReg, lsl #5
+    *    subw resultReg, dataReg, resultReg
+    *    b.gt residual
+    * done:
+    *    ret
+    *    .align 3
+    * multiplier:
+    *    .dword 0x3c10000745f   ; 31^2, 31^3
+    *    .dword 0x10000001f     ;    1, 31
+    * multiplier4:
+    *    .dword 0xe1781000e1781 ; 31^4
+    *    .dword 0xe1781000e1781
+    *
+    */
+
+   /*
+    * We omit to evaluate the second child because it is always 0.
+    */
+   TR::Node *arrayNode = node->getFirstChild();
+   TR::Node *lengthNode = node->getThirdChild();
+   TR::Register *arrayReg = cg->evaluate(arrayNode);
+   TR::Register *savedLengthReg = cg->evaluate(lengthNode);
+   TR_ARM64ScratchRegisterManager *srm = cg->generateScratchRegisterManager(12);
+   TR::Register *dataAddrReg = (arrayNode->getReferenceCount() > 1) ? srm->findOrCreateScratchRegister() : arrayReg;
+   TR::Register *lengthReg = (lengthNode->getReferenceCount() > 1) ? srm->findOrCreateScratchRegister() : savedLengthReg;
+   TR::Compilation *comp = cg->comp();
+   if (comp->getOptions()->enableDebugCounters())
+      {
+      cg->generateDebugCounter(TR::DebugCounter::debugCounterName(comp, "cg.StringHashCode/(%s)/%s",
+                                                                        comp->signature(),
+                                                                        comp->getHotnessName()), *srm);
+      }
+   TR::Register *resultReg = cg->allocateRegister();
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addimmx, node, dataAddrReg, arrayReg, TR::Compiler->om.contiguousArrayHeaderSizeInBytes());
+   loadConstant32(cg, node, 0, resultReg);
+   generateCompareImmInstruction(cg, node, lengthReg, (isCompressed ? 16 : 8), false);
+   TR::LabelSymbol *residualLabel = generateLabelSymbol(cg);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, residualLabel, TR::CC_LT);
+   if (comp->getOptions()->enableDebugCounters())
+      {
+      cg->generateDebugCounter(TR::DebugCounter::debugCounterName(comp, "cg.StringHashCode/(%s)/%s:long",
+                                                                        comp->signature(),
+                                                                        comp->getHotnessName()), *srm);
+      }
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::subimmw, node, lengthReg, lengthReg, (isCompressed ? 16 : 8));
+   TR::Register *multiplierReg = srm->findOrCreateScratchRegister(TR_VRF);
+   {
+      static uint32_t multiplier[] = { 31*31*31, 31*31, 31, 1 };
+      auto snippet = cg->findOrCreateConstantDataSnippet(node, multiplier, 16);
```

Comment:
Please add a comment for the magic number 16, or replace the number by `sizeof(multiplier)`, I guess.

Reply:
Replaced the number with `sizeof(multiplier)`.